sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -351,6 +351,9 @@ final class DataFrameWriter private[sql](df: DataFrame) {
    * Inserts the content of the [[DataFrame]] to the specified table. It requires that
    * the schema of the [[DataFrame]] is the same as the schema of the table.
    *
+   * Note: Unlike `saveAsTable`, `insertInto` ignores the column names. If the column order or names
+   * doesn't match the table's, you may get unexpected results.
```

Comment:
How about we explicitly say that we use position-based resolution?

For example, if we have a table `t1` with columns `c: int, d: int`, `df.select("a", "b").insertInto("t1")` will insert `a`'s data to column `c` and `b`'s data to column `d`?